File: setup-config.py
#!/usr/bin/env python3
"""
One-time setup: writes config.js with your Supabase URL, anon key, and admin email.
Run from project root. Your values stay in your terminal only.

Usage:
  python3 setup-config.py
"""

import os

def main():
    print("Enter your Supabase values (from Dashboard → Project Settings → API).\n")
    url = (os.environ.get("SUPABASE_URL") or input("SUPABASE_URL (e.g. https://xxxx.supabase.co): ")).strip()
    key = (os.environ.get("SUPABASE_ANON_KEY") or input("SUPABASE_ANON_KEY (anon public key): ")).strip()
    email = (os.environ.get("MASTER_EMAIL") or input("MASTER_EMAIL (your admin email): ")).strip()

    if not url or not key or not email:
        print("Missing a value. Run again and fill all three.")
        return

    # Escape for use inside JavaScript strings (backslashes and quotes)
    def js(s):
        return s.replace("\\", "\\\\").replace("'", "\\'").replace("\n", "\\n").replace("\r", "\\r")

    content = """/**
 * Optional config for Today's Daily Battle.
 * Add config.js to .gitignore — do not commit keys.
 */
window.TDB_CONFIG = {
  SUPABASE_URL: '%s',
  SUPABASE_ANON_KEY: '%s',
  MASTER_EMAIL: '%s',
  MASTER_EMAILS: ['%s'],
  WALKTHROUGH_VIDEO_URL: '',
  ERROR_REPORT_URL: ''
};
""" % (js(url), js(key), js(email), js(email))

    with open(os.path.join(os.path.dirname(__file__), "config.js"), "w") as f:
        f.write(content)

    print("config.js written. Reload the site and try signing in.")

if __name__ == "__main__":
    main()

File: setup-config.js
#!/usr/bin/env node
/**
 * One-time setup: writes config.js with your Supabase URL and anon key.
 * Admin access is not configured here — set app_metadata.role = admin in Supabase Dashboard.
 * Run from project root. Your values stay in your terminal only (never paste them in chat).
 *
 * Usage:
 *   node setup-config.js
 *
 * Or with env vars (paste in terminal, not in chat):
 *   SUPABASE_URL="https://xxxx.supabase.co" SUPABASE_ANON_KEY="eyJ..." node setup-config.js
 */

const fs = require('fs');
const path = require('path');

const configPath = path.join(__dirname, 'config.js');

const url = process.env.SUPABASE_URL || '';
const key = process.env.SUPABASE_ANON_KEY || '';

if (url && key) {
  writeConfig(url, key);
  console.log('config.js written successfully. Reload the site and try signing in.');
  process.exit(0);
}

const readline = require('readline');
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

function ask(q) {
  return new Promise((resolve) => rl.question(q, resolve));
}

(async () => {
  console.log('Enter your Supabase values (from Dashboard → Project Settings → API).\n');
  const u = (process.env.SUPABASE_URL || await ask('SUPABASE_URL (e.g. https://xxxx.supabase.co): ')).trim();
  const k = (process.env.SUPABASE_ANON_KEY || await ask('SUPABASE_ANON_KEY (anon public key): ')).trim();
  rl.close();
  if (!u || !k) {
    console.log('Missing SUPABASE_URL or SUPABASE_ANON_KEY. Run again.');
    process.exit(1);
  }
  writeConfig(u, k);
  console.log('config.js written. Set admin via Supabase app_metadata if needed; see CONFIG.md.');
})();

function writeConfig(supabaseUrl, anonKey) {
  const content = `/**
 * Optional config for Today's Daily Battle.
 * Add config.js to .gitignore — do not commit keys.
 */
window.TDB_CONFIG = {
  SUPABASE_URL: ${JSON.stringify(supabaseUrl)},
  SUPABASE_ANON_KEY: ${JSON.stringify(anonKey)},
  WALKTHROUGH_VIDEO_URL: '',
  ERROR_REPORT_URL: ''
};
`;
  fs.writeFileSync(configPath, content, 'utf8');
}
